internal/mfaportal/resources/frontend/src/App.vue

```diff
@@ -88,113 +88,136 @@ function getMFAPath(): string {
   return "/selection"
 }
 
+function isMfaMethodOidc(userMfaMethod: string): boolean {
+  return userMfaMethod === "oidc";
+}
+
+
+/*
+This function ensures navigation only happens when the router is fully ready,
+preventing navigation errors or hangs when pushing routes after async state changes.
+Specifically, it fixes the issue where closing and reopening the tab after deauthorizing
+would not redirect correctly on reauthorization.
+*/
+async function safeRouterPush(path: string) {
+  await router.isReady();
+  await router.push(path).catch((err) => {
+    console.error("Error detected when trying to redirect:", err);
+  });
+}
+
 async function initialRouting() {
   try {
-    await nextTick()
+    await nextTick();
 
     if (info.isDeviceLocked || info.isAccountLocked) {
-      console.log("determined locked")
-      router.push("/locked")
-      notify("VPN Locked", "Your device has been locked. Please contact help")
-      return
+      console.log("determined locked");
+      await safeRouterPush("/locked");
+      notify("VPN Locked", "Your device has been locked. Please contact help");
+      return;
     }
 
-    if (info.isAuthorised) {
-      console.log("determined authed")
-      router.push("/success")
-      return
+    if (!info.isRegistered) {
+      await safeRouterPush(getMFAPath());
+      return;
+    } else {
+      console.log("Registered checking if it's authorized");
+      if (!info.isAuthorised) {
+        console.log("not authorised, redirecting to authorise aka connect the tunnel");
+        if (isMfaMethodOidc(info.selectedMFAMethod)) {
+          await safeRouterPush("/login");
```

Comment:
This doesnt actually move us any closer to having the logout functionality work for OIDC logins. This just replaces the login flow with something that completely subverts the security properties of wag